tools/autograd/gen_variable_type.py

```diff
@@ -495,6 +495,8 @@ def get_suffix(dynamic_type, is_nullable):
             return '_opt'
         elif dynamic_type == 'IndexTensor':
             return '_long'
+        elif dynamic_type == 'IntegerTensor':
```

Comment:
Can someone explain to me where we are getting these names from? Why aren't we using names consistent with the public facing PyTorch API?

Reply:
I believe these names originated from the cwrap declarations.  Now that aten has it's own declarations we could and should rationalize these.

Reply:
The `IntegerTensor` I believe was only necessary because of the behavior of `LookupTable` from `nn`, which returned [counts as `int` instead of `long`](https://github.com/pytorch/pytorch/blob/master/aten/src/THNN/generic/LookupTable.c#L30). The original motivation was to have a single header that could work for both CPU and CUDA, so we could write the headers with `THTensor` and `THIndexTensor` and they would be replaced by the corresponding types depending on the device.
Some discussion about it can be found in https://github.com/torch/nn/pull/549